enaml/core/parser/parser36.py

```diff
@@ -96,21 +97,35 @@ def p_annassign(self, p):
     # that in "(f'{a}' '{r}')" we must not format r.
     def p_atom10(self, p):
         ''' atom : atom_string_list '''
-        m = ast.parse('(' + ' '.join(p[1]) + ')')
+        err_msg = ''
+        try:
+            m = ast.parse('(' + ' '.join([s for s, _ in p[1]]) + ')')
+        except SyntaxError as e:
+            err_msg = e.args[0]
+            for s, lineno in p[1]:
+                try:
+                    m = ast.parse(s)
+                except SyntaxError:
+                    break
+
+        if err_msg:
```

Comment:
Is the separate `err_msg = ''` and if statement needed? Also it may be good to add a comment why the for loop is there.

Reply:
The if avoids Python giving a triple level exception (exception during error handling). Actually I could apply something similar in `parse` to avoid having multiple level at all.